File: clases 25-11-2025/Tipos de clases c#/clase normal con interfaz/Curso.cs
public class Curso : ICurso
{
    public string Nombre { get; set; } = "sin nombre";
    public int Duracion { get; set; } = 0;

    public Curso() { }

    public void InfoCurso()
    {
        Console.WriteLine($"Curso: {Nombre}, Duración: {Duracion} horas");
    }
}

File: clases 25-11-2025/Tipos de clases GO/clase_normal_con_interfaz/Curso.go
package main

import "fmt"

type ICurso interface {
	InfoCurso()
}

type Curso struct {
	nombre   string
	duracion int
}

func NewCurso() ICurso {
	return Curso{nombre: "sin nombre", duracion: 0}
}

func (c Curso) InfoCurso() {
	fmt.Printf("Curso: %s, Duración: %d horas\n", c.nombre, c.duracion)
}
